stringvalidator/format.go

```diff
@@ -0,0 +1,108 @@
+ *
+ * This software is distributed under the MPL-2.0 license.
+ * the text of which is available at https://www.mozilla.org/en-US/MPL/2.0/
+ * or see the "LICENSE" file for more details.
+ */
+
+package stringvalidator
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+
+	formatTypes "github.com/orange-cloudavenue/terraform-plugin-framework-validators/stringvalidator/format"
+)
+
+var _ validator.String = formatValidator{}
+
+const (
+	FormatIsBase64 FormatValidatorType = "is_base64"
+	FormatIsUUIDv4 FormatValidatorType = "is_uuid_v4"
+	FormatIsURN    FormatValidatorType = "is_urn"
+)
+
+var formatTypesFunc = map[FormatValidatorType]func() validator.String{
+	FormatIsBase64: formatTypes.IsBase64,
+	FormatIsUUIDv4: formatTypes.IsUUIDv4,
+	FormatIsURN:    formatTypes.IsURN,
+}
+
+type FormatValidatorType string
+
+type formatValidator struct {
+	FormatTypes []FormatValidatorType
+}
+
+// Description describes the validation in plain text formatting.
+func (validatorFormat formatValidator) Description(ctx context.Context) string {
+	description := ""
+
+	if len(validatorFormat.FormatTypes) == 0 {
+		description += "invalid configuration"
+	}
+
+	switch {
+	case len(validatorFormat.FormatTypes) > 1:
+		description += "The value must respect the following rules : "
+	case len(validatorFormat.FormatTypes) == 1:
+		description += "The value must respect the following rule : "
+	}
+
+	for i, formatType := range validatorFormat.FormatTypes {
+		if i == len(validatorFormat.FormatTypes)-1 {
+			description += formatTypesFunc[formatType]().Description(ctx)
+		} else {
+			description += formatTypesFunc[formatType]().Description(ctx) + ", "
+		}
+	}
+
+	return description
+}
+
+// MarkdownDescription describes the validation in Markdown formatting.
+func (validatorFormat formatValidator) MarkdownDescription(ctx context.Context) string {
+	return validatorFormat.Description(ctx)
+}
+
+// Validate performs the validation.
+func (validatorFormat formatValidator) ValidateString(
+	ctx context.Context,
+	request validator.StringRequest,
+	response *validator.StringResponse,
+) {
+	if request.ConfigValue.IsNull() || request.ConfigValue.IsUnknown() {
+		return
+	}
+
+	if len(validatorFormat.FormatTypes) == 0 {
+		response.Diagnostics.AddAttributeError(
+			request.Path,
+			"No configuration provided",
+			fmt.Sprintf("invalid value: %s", request.ConfigValue.String()),
+		)
+		return
+	}
+
+	for _, formatType := range validatorFormat.FormatTypes {
+		if formatTypesFunc[formatType] == nil {
+			response.Diagnostics.AddAttributeError(
+				request.Path,
+				"Invalid configuration",
+				fmt.Sprintf("invalid value: %s", request.ConfigValue.String()),
+			)
+			return
+		}
+		formatTypesFunc[formatType]().ValidateString(ctx, request, response)
+	}
+}
+
+func Format(formatTypes []FormatValidatorType) validator.String {
+	return &formatValidator{
+		FormatTypes: formatTypes,
+	}
+}
```

Comment:
Add parameter to define if multiple formats are used the comparator is OR or AND